docs/guides/deploy-hyperlane-local-agents.mdx

```diff
@@ -0,0 +1,397 @@
+
+# Create the directory
+mkdir -p $VALIDATOR_SIGNATURES_DIR
+```
+
+:::
+
+### Configure
+
+There are numerous parameters that validators can be configured with. For this guide, we are concerned with just a handful:
+
+| Parameter                 | Description                                                         |
+| ------------------------- | ------------------------------------------------------------------- |
+| `--db`                    | Path for writing persistent data to disk.                           |
+| `--originChainName`       | Name of the chain being validated (e.g. `ethereum`).                |
+| `--checkpointSyncer.type` | Set to `localStorage` for this guide.                               |
+| `--checkpointSyncer.path` | Path to local directory where validator signatures will be written. |
+| `--validator.key`         | Your validator's hexadecimal private key.                           |
+
+:::info
+
+Make sure the validator key corresponds to the address provided when setting up your MultisigIsmConfig. Otherwise, the Multisig ISM you deployed in the previous step will not be able to verify messages sent from your chain.
+
+:::
+
+To learn more about all the parameters you can change, read the [agent configuration reference](../operate/config-reference.mdx).
+
+<Tabs groupId="docker">
+<TabItem value="docker" label="Using Docker">
+
+**Update agent config**
+
+Unless you are running Docker on Linux, you will also need to update the agent configuration for your network. This is because Docker does not support the [`host` network mode](https://docs.docker.com/network/drivers/host/) on Mac, Windows or Windows Server.
+
+To do this, navigate to the agent-configuration at `$CONFIG_FILES` and replace all instances of "localhost" or "127.0.0.1" in to `host.docker.internal`. For example:
+
+```json
+...
+"localnet1": {
+  ...
+  "rpcUrls": [
+    {
+      // "http": "http://localhost:8545"
+      // "http": "http://127.0.0.1:8545"
+      "http": "http://host.docker.internal:8545"
+    }
+  ],
+  ...
+},
+...
+```
+
+**Mounting directories**
+
+Running with Docker adds an extra layer of complexity because config files need to be accessible from within the Docker container, and validator signatures need to be accessible from outside of the container. This can be solved by mounting directories on your file system into the container.
+
+In the arguments below, we:
+
+1. Set the `$CONFIG_FILES` environment variable to a fixed path within the container.
+1. Mount the agent config file to this fixed path and making it readonly.
+1. Mount the persistent data directory at a fixed path within the container.
+1. Mount the validator signatures directory to a fixed path within the container.
+
+```sh
+...
+-e CONFIG_FILES=/config/agent-config.json \
+--mount type=bind,source=$CONFIG_FILES,target=/config/agent-config.json,readonly \
+--mount type=bind,source="$(pwd)"/hyperlane_db_validator_<your_chain_name>,target=/hyperlane_db \
```

Comment:
does the `.../hyperlane_db_validator_<your_chain_name>` path need to be created beforehand? Same question for the relayer db path

Reply:
yes good catch! had in my notes but forgot to add to doc